core/types/block_ext_test.go

```diff
@@ -106,3 +111,410 @@ func assertDifferentPointers[T any](t *testing.T, a *T, b any) {
+		ExtData: ptrTo([]byte{2}),
+	}
+	extras.Body.Set(wantBody, wantExtra)
+
+	assert.Equal(t, wantBody, gotBody)
+	gotExtra := extras.Body.Get(gotBody)
+	assert.Equal(t, wantExtra, gotExtra)
+}
+
+func TestBlockExtraRLP(t *testing.T) {
+	t.Parallel()
+
+	header := &Header{ParentHash: common.Hash{1}}
+	headerExtras := &HeaderExtra{ExtDataHash: common.Hash{2}}
+	SetHeaderExtra(header, headerExtras)
+
+	testTx := NewTransaction(3, common.Address{4}, big.NewInt(2), 3, big.NewInt(4), []byte{5})
+	txs := []*Transaction{testTx}
+
+	uncle := &Header{ParentHash: common.Hash{6}}
+	uncleExtra := &HeaderExtra{ExtDataHash: common.Hash{7}}
+	SetHeaderExtra(uncle, uncleExtra)
+	uncles := []*Header{uncle}
+
+	receipts := []*Receipt{{PostState: []byte{8}}}
+
+	block := NewBlock(header, txs, uncles, receipts, stubHasher{})
+
+	withdrawals := []*ethtypes.Withdrawal{{Index: 9}} // ignored
+	block = block.WithWithdrawals(withdrawals)
+
+	extra := &BlockBodyExtra{
+		Version: 10,
+		ExtData: ptrTo([]byte{11}),
+	}
+	extras.Block.Set(block, extra)
+
+	b, err := rlp.EncodeToBytes(block)
+	require.NoError(t, err)
+
+	gotBlock := new(Block)
+	err = rlp.DecodeBytes(b, gotBlock)
+	require.NoError(t, err)
+
+	// Check transactions in the following because unexported fields
+	// size and time are set by the RLP decoder.
+	gotTxs := gotBlock.Transactions()
+	require.Len(t, gotTxs, 1)
+	gotTx := gotTxs[0]
+	gotTxBin, err := gotTx.MarshalBinary()
+	require.NoError(t, err)
+	wantTxBin, err := testTx.MarshalBinary()
+	require.NoError(t, err)
+	assert.Equal(t, wantTxBin, gotTxBin)
+
+	wantHeader := &Header{
+		ParentHash: common.Hash{1},
+		Extra:      []byte{}, // field set by RLP decoder
+	}
+	wantHeaderExtras := &HeaderExtra{ExtDataHash: common.Hash{2}}
+	SetHeaderExtra(wantHeader, wantHeaderExtras)
+
+	wantUncle := &Header{
+		ParentHash: common.Hash{6},
+		Difficulty: new(big.Int), // field set by RLP decoder
+		Number:     new(big.Int), // field set by RLP decoder
+		Extra:      []byte{},     // field set by RLP decoder
+	}
+	wantUncleExtra := &HeaderExtra{ExtDataHash: common.Hash{7}}
+	SetHeaderExtra(wantUncle, wantUncleExtra)
+	wantUncles := []*Header{wantUncle}
+
+	wantBlock := NewBlock(wantHeader, gotTxs /* checked above */, wantUncles, receipts, stubHasher{})
+	wantExtra := &BlockBodyExtra{
+		Version: 10,
+		ExtData: ptrTo([]byte{11}),
+	}
+	extras.Block.Set(wantBlock, wantExtra)
+	_ = wantBlock.Size() // set block cached unexported size field
+
+	assert.Equal(t, wantBlock, gotBlock)
+	gotExtra := extras.Block.Get(gotBlock)
+	assert.Equal(t, wantExtra, gotExtra)
+}
+
+// TestBlockBody tests the [BlockBodyExtra.Copy] method is implemented correctly.
+func TestBlockBody(t *testing.T) {
+	t.Parallel()
+
+	header := &Header{ParentHash: common.Hash{1}}
+	tx := NewTransaction(0, common.Address{1}, big.NewInt(2), 3, big.NewInt(4), []byte{5})
+	txs := []*Transaction{tx}
+	uncles := []*Header{{ParentHash: common.Hash{6}}}
+	receipts := []*Receipt{{PostState: []byte{7}}}
```

Comment:
If this is only a test of `BlockBodyExtra.Copy` then the rest of these fields are irrelevant and only distract from the test. The method doesn't have access to the `Block` so there's no concern about it mutating the fields.

Reply:
Tests refactored and this is no longer needed, resolving this!